.github/workflows/ci.yml

```diff
@@ -7,7 +7,33 @@ on:
 permissions: read-all
 
 jobs:
+  check_merge_conflicts:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Check for Merge Conflicts
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const prNumber = context.payload.pull_request?.number;
+            if (!prNumber) {
+              console.log("Not a pull request, skipping merge conflict check.");
+              return;
+            }            
+            const { data: pr } = await github.rest.pulls.get({
```

Comment:
I don't know we can use this endpoint. I know from experience that github takes a while to figure out if something can be merged or not. Isn't the error message from the checkout action enough?

Reply:
I'm totally down to find another way of accomplishing this--you hit on the head my biggest concern about Github taking too long to settle (we don't want to have maintainers constantly having to kick CI). Can you give a bit more color to what you're thinking with the checkout action?

As for the rest, I don't really think having an extra action (especially if it prevents other bogus runs) is a big deal, and I'd hope that if development pace picks up we'd see more potential conflicts.